pyEpiabm/pyEpiabm/tests/test_sweep/test_spatial_sweep.py

```diff
@@ -45,77 +48,91 @@ def test__call__(self, mock_inf, mock_force, mock_poisson, mock_dist,
         mock_inf.return_value = 10.0
         mock_force.return_value = 100.0
         time = 1
-        Parameters.instance().infection_radius = 5000
+        Parameters.instance().infection_radius = 1000
 
+        test_pop = self.pop
```

Comment:
@rcw5890 do we need a deep copy of the population here for stability, or should this suffice?

EDIT: They give the same hash for me (i.e. point to the same object) so I believe this is not sufficient and you will need a deep copy - otherwise changes to test_pop will also affect self.pop. Richard might be able to advise on best practice here though

Reply:
@Elizabeth-Hayman @KCGallagher I agree we ought to get a new copy of the population. Unfortunately, copy.deepcopy causes weird problems possibly related to logging, I'm guessing you saw the same thing at some point ?

The better option may be to convert `setUpClass` (called once before running all tests in the class) to `setUp` (called before each test). This will give us a fresh population for each test.

As written `test_do_infection_event` seems to depend on some of the changes to the population that happen in the previous test, so that will require some minor changes !

Reply:
I think I understand this and have made the relevant changes

Reply:
Ahh that sounds the ideal solution (and may even be worth using elsewhere in the code too) - I didn't realise there was this difference between `setUp` and `setUpClass`. We won't need the `super` line in the setup now as we're not overwriting the `setUpClass` from my custom parent, but it will be worth confirming that the logger doesn't print (I haven't tested this scenario but it should still work)